docs/examples/gnoi/gnoicode.md

```diff
@@ -0,0 +1,203 @@
+---
+layout: default
+title: "gNOI in Golang"
+date: 2022-01-06 10:07:00 --0600
+categories:
+---
+## gNOI review
+
+These are a few examples leveraging golang and gNOI using the gNOI protobuf specifically the [System service](https://github.com/openconfig/gnoi/blob/master/system/system.proto)
+
+The following examples show how to leverage the [ping rpc](https://github.com/openconfig/gnoi/blob/master/system/system.proto#L41) and [traceroute rpc](https://github.com/openconfig/gnoi/blob/master/system/system.proto#L47)
+
+All code can be found within the [src directory under gnoi](https://github.com/aristanetworks/openmgmt/tree/main/src/gnoi)
+
+We will be leveraging the gNOI [godocs]((https://pkg.go.dev/github.com/openconfig/gnoi)) which module can be imported as github.com/openconfig/gnoi
+
+Each one of the examples has the following constants which can be changed based on the environment.
+
+```golang
+const (
+	username    = "admin"
+	password    = "admin"
+	target      = "172.20.20.2:6030"
+	destination = "1.1.1.1"
+	timeOut     = 5
+)
+```
+
+## Test Device configuration
+<br>
+
+```shell
+management api gnmi
+   transport grpc default
+   provider eos-native
+```
+
+### cd into src/gnoi
+<br>
+
+```shell
+cd src/gnoi
+```
+<br>
+
+The directory should include the go.mod/go.sum for the correct packages so nothing needs installed on the system at the current time.
+<br>
+<br>
+
+```shell
+ls
+go.mod  go.sum  pinggnoi.go  traceroutegnoi.go
+```
+
+### Run ping.go
+
+#### src/gnoi/ping.go
+
+<details><summary>Reveal output</summary>
+<p>
+
+```golang
```

Comment:
i'm wondering if we should embed these so that they're automatically aligned and changes only need to be made in  a single location. 

i think we can do this by using mkdocs snippets extension.

Reply:
Sounds good to me.

Reply:
sounds good to me.

Reply:
to be clear - i was referring to use of `--8<-- " src/gnoi/traceroute.go"` here which would include your go code into the markdown file without requiring that you update things in both locations.  

snippets need to be enabled in the `mkdocs.yml` file as well.  
``` 
# markdown extensions
markdown_extensions:
  - pymdownx.snippets
```

this would also address a lot of the markdown linting issues.